packages/api/src/server/search/search.controller.ts

```diff
@@ -0,0 +1,19 @@
+import { Controller, Get, Param, UseGuards } from '@nestjs/common';
+import { ApiCreatedResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { AuthenticationGuard } from 'src/server/auth/auth.guard';
+import { Rover } from './rover.response.dto';
+import { SearchService } from './service/search.service';
+
+@Controller('search')
+@ApiTags('Search')
+@UseGuards(AuthenticationGuard)
+export class SearchController {
```

Comment:
Is this class specifically related to rover?

Reply:
No actually, This is the generalises search controller, currently we have the Usersearch on this.